cli/tools/registry/pm/outdated.rs

```diff
@@ -0,0 +1,607 @@
+use crate::args::Flags;
+use crate::args::OutdatedFlags;
+use crate::factory::CliFactory;
+use crate::file_fetcher::FileFetcher;
+use crate::jsr::JsrFetchResolver;
+use crate::npm::NpmFetchResolver;
+use crate::tools::registry::pm::deps::DepKind;
+
+use super::deps::Dep;
+use super::deps::DepManager;
+use super::deps::DepManagerArgs;
+use super::deps::PackageLatestVersion;
+
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord)]
+struct OutdatedPackage {
+  kind: DepKind,
+  latest: String,
+  semver_compatible: String,
+  current: String,
+  name: String,
+}
+
+#[allow(clippy::print_stdout)]
+fn print_outdated_table(packages: &[OutdatedPackage]) {
+  const HEADINGS: &[&str] = &["Package", "Current", "Update", "Latest"];
+
+  let mut longest_package = 0;
+  let mut longest_current = 0;
+  let mut longest_update = 0;
+  let mut longest_latest = 0;
+
+  for package in packages {
+    let name_len = package.kind.scheme().len() + 1 + package.name.len();
+    longest_package = longest_package.max(name_len);
+    longest_current = longest_current.max(package.current.len());
+    longest_update = longest_update.max(package.semver_compatible.len());
+    longest_latest = longest_latest.max(package.latest.len());
+  }
+
+  let package_column_width = longest_package.max(HEADINGS[0].len()) + 2;
+  let current_column_width = longest_current.max(HEADINGS[1].len()) + 2;
+  let update_column_width = longest_update.max(HEADINGS[2].len()) + 2;
+  let latest_column_width = longest_latest.max(HEADINGS[3].len()) + 2;
+
+  let package_fill = "─".repeat(package_column_width);
+  let current_fill = "─".repeat(current_column_width);
+  let update_fill = "─".repeat(update_column_width);
+  let latest_fill = "─".repeat(latest_column_width);
+
+  println!("┌{package_fill}┬{current_fill}┬{update_fill}┬{latest_fill}┐");
+  println!(
+    "│ {}{} │ {}{} │ {}{} │ {}{} │",
+    colors::intense_blue(HEADINGS[0]),
+    " ".repeat(package_column_width - 2 - HEADINGS[0].len()),
+    colors::intense_blue(HEADINGS[1]),
+    " ".repeat(current_column_width - 2 - HEADINGS[1].len()),
+    colors::intense_blue(HEADINGS[2]),
+    " ".repeat(update_column_width - 2 - HEADINGS[2].len()),
+    colors::intense_blue(HEADINGS[3]),
+    " ".repeat(latest_column_width - 2 - HEADINGS[3].len())
+  );
+  for package in packages {
+    println!("├{package_fill}┼{current_fill}┼{update_fill}┼{latest_fill}┤",);
+
+    print!(
+      "│ {:<package_column_width$} ",
+      format!("{}:{}", package.kind.scheme(), package.name),
+      package_column_width = package_column_width - 2
+    );
+    print!(
+      "│ {:<current_column_width$} ",
+      package.current,
+      current_column_width = current_column_width - 2
+    );
+    print!(
+      "│ {:<update_column_width$} ",
+      package.semver_compatible,
+      update_column_width = update_column_width - 2
+    );
+    println!(
+      "│ {:<latest_column_width$} │",
+      package.latest,
+      latest_column_width = latest_column_width - 2
+    );
```

Comment:
I still think that printing how old a version is (in hours or days) is useful here.

Reply:
There were two issues with that:
- deno npm is missing the time field from this struct: https://docs.rs/deno_npm/latest/deno_npm/registry/struct.NpmPackageInfo.html (though I guess I could do some custom deserialization)
- jsr doesn't expose that info, so it would only be available for npm packages, which feels not great

Reply:
That's fine then, we can skip it for now and work on it in a follow up